### PR TITLE
Fix new document's name translation https://github.com/FreeCAD/FreeCAD-translations/issues/124 

### PR DESCRIPTION
**Pull request by Kuzma30**
This solve such things:
1.  If you create new document, it show "Unnamed" name without translations. Solved.
2.  If you open .step file FreeCAD create new document with "Unnamed" name without translations. Solved.
3.  If you open .iges file FreeCAD create new document with "Unnamed" name without translations. Solved.
4.  Add similar translation ability to AppFemPy, AppMeshPy and AppPartPy.

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [ ]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6825">pipeline status</a> for feature branch [PR_6825](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6825). Pipeline [529847895](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/529847895) was triggered at [8c2026e](https://github.com/FreeCAD/FreeCAD/pull/6825/commits/8c2026e1312f20fd9605f3c8ee4428fc4eff6469). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6825">pipeline status</a> for feature branch [PR_6825](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6825). Pipeline [529849407](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/529849407) was triggered at [ba4871c](https://github.com/FreeCAD/FreeCAD/pull/6825/commits/ba4871c09e70a1a99f623f352b09b0591c5fd1d2). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by luzpaz** [comment]
please remove the `.ts` and `.qm` files

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6825">pipeline status</a> for feature branch [PR_6825](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6825). Pipeline [529861661](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/529861661) was triggered at [5332cfc](https://github.com/FreeCAD/FreeCAD/pull/6825/commits/5332cfc668f6e5a88e77204db83ea82888d3195a). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by yorikvanhavre** [comment]
TBH I'm a little unsure to introduce this kind of change so close to the release, as it might trigger more bugs, there might still be some functions here and there that rely on an ASCII doc name. I'd vote to postpone to after 0.20 release.

**Comment by Kuzma30** [comment]
Please mark this for v0.21

File: src/App/Application.cpp [hunk 1]
```diff
@@ -435,9 +436,11 @@ void Application::renameDocument(const char *OldName, const char *NewName)
 
 Document* Application::newDocument(const char * Name, const char * UserName, bool createView, bool tempDoc)
 {
+    QString L10nName = QObject::tr("Unnamed");
     // get a valid name anyway!
     if (!Name || Name[0] == '\0')
-        Name = "Unnamed";
+        Name = qUtf8Printable(L10nName);
```
Comment (chennes) [inline src/App/Application.cpp:442]: You cannot store the pointer created by `qUtf8Printable`, it falls out of scope immediately following the statement.

**Review by chennes — CHANGES_REQUESTED** [review]
See comment regarding storage of result of `qUtf8Printable` -- that chunk of code needs to be refactored so that we don't have to store that `char *`.

**Comment by Kuzma30** [comment]
@chennes Fix it. What do you think about https://github.com/FreeCAD/FreeCAD/pull/6825#issuecomment-1115757651?
Without .ts it will not work I think.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6825">pipeline status</a> for feature branch [PR_6825](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6825). Pipeline [571389674](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/571389674) was triggered at [c45eb02](https://github.com/FreeCAD/FreeCAD/pull/6825/commits/c45eb02d9cdda5170b84ed7b6e8839705ad0055b). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by chennes** [comment]
Our normal process is to have @yorikvanhavre deal with running `updatets.py` to generate the TS files and upload to CrowdIn. I don't think he minds help, but it's not necessary. We typically separate the code updates from the translation file updates, and even with the TS files, we still have to wait for Yorik to do a CrowdIn upload/download cycle.

**Comment by Kuzma30** [comment]
> Our normal process is to have @yorikvanhavre deal with running `updatets.py` to generate the TS files and upload to CrowdIn. I don't think he minds help, but it's not necessary. We typically separate the code updates from the translation file updates, and even with the TS files, we still have to wait for Yorik to do a CrowdIn upload/download cycle.

More info. This file (App.ts) is new and I create new file App.qrc too. I think I can add generation of App.ts to updatets.py script.

P.S. Waiting for enabling crowdin github integration ...

**Comment by chennes** [comment]
Sorry, I didn't see that they were new files. Is there a corresponding modification to `updatets.py` to generate them in the first place?

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6825">pipeline status</a> for feature branch [PR_6825](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6825). Pipeline [571867734](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/571867734) was triggered at [2ad3068](https://github.com/FreeCAD/FreeCAD/pull/6825/commits/2ad30689c3b36d9d18c6dc1e3f0ec723bab63291). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by Kuzma30** [comment]
@chennes I change updatets.py. But I don`t know what to do with App.qrc. In similar PR https://github.com/FreeCAD/FreeCAD/pull/7067 I think we need .qrc modification too. But this need creating .ts files for another languages (all non English) by hand.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6825">pipeline status</a> for feature branch [PR_6825](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6825). Pipeline [571869600](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/571869600) was triggered at [98781cc](https://github.com/FreeCAD/FreeCAD/pull/6825/commits/98781cc4345bedc4bd7d6958920e7af6bd483d46). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by chennes** [comment]
We can do it in two stages: just add the main TS file now, then after the next CrowdIn cycle we can add the language files.

**Comment by yorikvanhavre** [comment]
I had a better look at this, and this question: Should we translate the document name? The name is an ASCII string, imagine if you are in chinese or arabic, your doc name will be something like _________. I think the Name should stay Unnamed, but the UserName is what should be translated.
And I'd prefer maybe not to do that in App but in Gui to keep App as translation-free as possible.

**Comment by Kuzma30** [comment]
We always has translation for document name. This PR related only first document name. First screenshot in
https://github.com/FreeCAD/FreeCAD-translations/issues/124

**Comment by Kuzma30** [comment]
> Still, I think the UserName is what should be translated, no? It would look OK on https://github.com/FreeCAD/FreeCAD-translations/issues/124 too

I delete tr() in App, change         Name = "Unnamed"; to Name ="UnnamedDoc" for test. After compiling I see first new document name as UnnamedDoc.
Need more research with UserName, need more time.

**Comment by Kuzma30** [comment]
@chennes @yorikvanhavre 
Another idea without .ts in App (I still have small problems) 
1. I make all call of NewDocument without parameters as we have in newDocument function setting value for internal name ("Unnamed"). Thus no need translate this in many code places.
2. I make parameter where is UserName will be stored.
3. This partially work. When there is now config files (I delete it) if you open Preferences, you can see New Document Name = Unnamed. Then if you exit from dialog without saving parameter and press New you get new document with empty name.
If you save parameters, then new document have correct name.
https://github.com/FreeCAD/FreeCAD/compare/master...Kuzma30:new_document

**Comment by Kuzma30** [comment]
@yorikvanhavre or @chennes or anybody of developers:) I need feedback about idea make parameter for new document's name. If it fine I continue on error fixing.

**Pull request closed without merging**